dnsdist: Unused lambda capture warnings and unused function warning in dnsdist-lua.cc

<!-- Tell us what is issue is about -->
 - Program: dnsdist <!-- delete the ones that do not apply -->
 - Issue type: Bug report

### Short description
compiler warning on dnsdist-lua.cc

### Environment
 - Operating system: Arch Linux
 - Software version: git (d26856a32e7506817c291c49d8eb5a60129dd751)
 - Software source: git

### Steps to reproduce
```
autoreconf -fi
CXX=clang++ CC=clang LD=lld ./configure
make
```

### Expected behaviour
No warnings

### Actual behaviour
```
dnsdist-lua.cc:1642:39: warning: lambda capture 'client' is not used [-Wunused-lambda-capture]
  g_lua.writeFunction("addDOHLocal", [client](const std::string& addr, boost::variant<std::string, std::vector<std::pair<int,std::string>>> certFiles, boost::variant<std::string, std::vector<std::pair<int,std::string>>> keyFiles, boost::optional<boost::variant<std::string, vector<pair<int, std::string> > > > urls, boost::optional<localbind_t> vars) {
                                      ^~~~~~
dnsdist-lua.cc:1756:39: warning: lambda capture 'client' is not used [-Wunused-lambda-capture]
  g_lua.writeFunction("addTLSLocal", [client](const std::string& addr, boost::variant<std::string, std::vector<std::pair<int,std::string>>> certFiles, boost::variant<std::string, std::vector<std::pair<int,std::string>>> keyFiles, boost::optional<localbind_t> vars) {
                                      ^~~~~~
dnsdist-lua.cc:112:13: warning: unused function 'loadTLSCertificateAndKeys' [-Wunused-function]
static bool loadTLSCertificateAndKeys(const std::string& context, std::vector<std::pair<std::string, std::string>>& pairs, boost::variant<std::string, std::vector<std::pair<int,std::string>>> certFiles, boost::variant<std::string, std::vector<std::pair<int,std::string>>> keyFiles)
            ^
3 warnings generated.
```

--- a/pdns/dnsdist-lua.cc
+++ b/pdns/dnsdist-lua.cc
@@ -1827,10 +1827,10 @@ static void setupLuaConfig(bool client, bool configCheck)
   });
 
   g_lua.writeFunction("addDOHLocal", [client](const std::string& addr, boost::optional<boost::variant<std::string, std::vector<std::pair<int,std::string>>>> certFiles, boost::optional<boost::variant<std::string, std::vector<std::pair<int,std::string>>>> keyFiles, boost::optional<boost::variant<std::string, vector<pair<int, std::string> > > > urls, boost::optional<localbind_t> vars) {
-#ifdef HAVE_DNS_OVER_HTTPS
     if (client) {
       return;
     }
+#ifdef HAVE_DNS_OVER_HTTPS
     setLuaSideEffect();
     if (g_configurationDone) {
       g_outputBuffer="addDOHLocal cannot be used at runtime!\n";
@@ -2023,9 +2023,10 @@ static void setupLuaConfig(bool client, bool configCheck)
       });
 
   g_lua.writeFunction("addTLSLocal", [client](const std::string& addr, boost::variant<std::string, std::vector<std::pair<int,std::string>>> certFiles, boost::variant<std::string, std::vector<std::pair<int,std::string>>> keyFiles, boost::optional<localbind_t> vars) {
-#ifdef HAVE_DNS_OVER_TLS
-        if (client)
+        if (client) {
           return;
+        }
+#ifdef HAVE_DNS_OVER_TLS
         setLuaSideEffect();
         if (g_configurationDone) {
           g_outputBuffer="addTLSLocal cannot be used at runtime!\n";

